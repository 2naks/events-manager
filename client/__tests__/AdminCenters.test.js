import ConnectedAdminCenters, { AdminCenters } from '../components/AdminCenters.jsx';

describe('AdminCenters Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      allCenters: {
        data: [{
          id: '1',
          name: 'City Hall',
          description: {
            substr: () => {}
          },
          location: {
            substr: () => {}
          },
          capacity: "500",
          price: "50000"
        }]
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    show: () => {},
    allCenters: {
      data: []
    }
  };

  describe('AdminCenters form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AdminCenters {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedAdminCenters /></Provider>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
