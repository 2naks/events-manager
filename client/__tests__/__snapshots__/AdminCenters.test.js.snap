// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminCenters Component AdminCenters form  should render self it should render for dumb component 1`] = `
<div
  className="panel centers"
  id="centers"
>
  <div
    className="container events"
    id="events"
  >
    <div
      className="row event-row"
    />
  </div>
</div>
`;
